crates/sui/src/sui_move/mod.rs

```diff
@@ -10,46 +11,34 @@ pub mod build;
 pub mod new;
 pub mod unit_test;
 
+#[derive(Parser)]
+pub enum Command {
+    Build(build::Build),
+    New(new::New),
+    Test(unit_test::Test),
+}
+
 pub fn execute_move_command(
-    package_path: PathBuf,
-    dump_bytecode_as_base64: bool,
+    package_path: Option<PathBuf>,
     build_config: BuildConfig,
-    command: PackageCommand,
+    command: Command,
 ) -> anyhow::Result<()> {
     match command {
-        PackageCommand::Build => {
-            build::execute(&package_path, dump_bytecode_as_base64, build_config)
-        }
-        PackageCommand::UnitTest {
-            instruction_execution_bound,
-            filter,
-            list,
-            num_threads,
-            report_statistics,
-            report_storage_on_error,
-            check_stackless_vm,
-            verbose_mode,
-            compute_coverage,
-        } => {
+        Command::Build(c) => c.execute(package_path, build_config),
+        Command::Test(c) => {
             let unit_test_config = UnitTestingConfig {
-                instruction_execution_bound,
-                filter,
-                list,
-                num_threads,
-                report_statistics,
-                report_storage_on_error,
-                check_stackless_vm,
-                verbose: verbose_mode,
+                instruction_execution_bound: c.test.instruction_execution_bound,
+                filter: c.test.filter.clone(),
+                list: c.test.list,
+                num_threads: c.test.num_threads,
+                report_statistics: c.test.report_statistics,
+                report_storage_on_error: c.test.report_storage_on_error,
+                check_stackless_vm: c.test.check_stackless_vm,
+                verbose: c.test.verbose_mode,
 
                 ..UnitTestingConfig::default_with_bound(None)
             };
-            let result = unit_test::execute(
-                &package_path,
-                dump_bytecode_as_base64,
-                build_config,
-                unit_test_config,
-                compute_coverage,
-            )?;
+            let result = c.execute(package_path, build_config, unit_test_config)?;
```

Comment:
The building of the unit test config should be done inside execute. And the. You should t need these clones

Reply:
We do add a custom flag to `UnitTestConfig` in crates/sui_framework/src/lib.rs and I am not sure how to do it without constructing the `UnitTestConfig` upfront:
```rust
move_cli::base::test::run_move_unit_tests(
        path,
        build_config,
        UnitTestingConfig {
            report_stacktrace_on_abort: true,
            ..config
        },
        natives::all_natives(MOVE_STDLIB_ADDRESS, SUI_FRAMEWORK_ADDRESS),
        compute_coverage,
        &mut std::io::stdout(),
    )
```